rust/core-lib/src/rpc.rs

```diff
@@ -513,16 +513,19 @@ impl<'de, T: Deserialize<'de>> Deserialize<'de> for EditCommand<T>
         let mut v = Value::deserialize(deserializer)?;
         let helper = InnerId::deserialize(&v).map_err(de::Error::custom)?;
         let InnerId { view_id } = helper;
-        // if params are empty, remove them
-        let remove_params = match v.get("params") {
-            Some(&Value::Object(ref obj)) => obj.is_empty(),
-            Some(&Value::Array(ref arr)) => arr.is_empty(),
-            Some(_) => return Err(de::Error::custom("'params' field, if present, must be object or array.")),
-            None => false,
-        };
 
-        if remove_params {
-            v.as_object_mut().map(|v| v.remove("params"));
+        if T::deserialize(v.clone()).is_err() {
```

Comment:
This is a little concerning, because it means we're always cloning the contents of each RPC we receive, which I suspect has measurable performance implications.

I think it would be okay to take this approach _only_ when the params are empty; in that case the clone will be cheap. In any case it's annoying that we can't somehow limit the clone to situations where deserialize fails; or alternately that we can't recover the original value in the failure case. 😞

Reply:
Yeah :(